pkgs/development/python-modules/mlx/default.nix

```diff
@@ -65,28 +72,38 @@ buildPythonPackage rec {
     ];
   };
 
-  nativeBuildInputs = [
+  build-system = [
     cmake
+    nanobind
+    setuptools
+  ];
+  dontUseCmakeConfigure = true;
+
+  nativeBuildInputs = [
     pybind11
     xcbuild
     zsh
     gguf-tools
     nlohmann_json
-    setuptools
```

Comment:
Also putting `setupttols` and `cmake` in `build-system` could be a separate commit.